recipe_recommender.mdx

```diff
@@ -0,0 +1,395 @@
+Rather than scouring the vast expanse of the internet, it narrows down your options to offer the most delectable
+recipe suggestions tailored to your preferences.
+
+# Core Components
+
+1. **OpenAI**: Used for generating embeddings, or vector representations, of recipe texts.
+2. **Pinecone**: A vector database service that efficiently stores these embeddings. Storing recipes as vectors will
+allow us to easily query the database for related recipes based on our food preferences.
+3. **PromptLayer**: A platform that enables collaborative prompt engineering by allowing teams and individuals to
+systematically track, debug, and optimize prompts for language models.
+
+# Setting Up the Environment
+
+To get started, visit the [GitHub repo](https://github.com/buckyroberts/Recipe-Recommender) for this project and
+download the [data files](https://github.com/buckyroberts/Recipe-Recommender/tree/main/data). Each one contains a
+recipe that we will use throughout the project.
+
+Next, we need to set up the API keys for the services we will be using.
+
+### OpenAI setup
+1. Create an API key at https://platform.openai.com/account/api-keys
+2. Save it as environment variable under `OPENAI_API_KEY`
+
+### Pinecone setup
+1. Create an API key at https://app.pinecone.io/
+2. Save it as environment variable under `PINECONE_API_KEY`
+
+### PromptLayer setup
+1. Create an API key at https://promptlayer.com/home
+2. Save it as environment variable under `PROMPTLAYER_API_KEY`
+
+# Diving into the Code
+
+Start by creating a `settings.py` file for our application level constants.
+<CodeGroup>
+```python settings.py
+from pathlib import Path
+
+BASE_DIR = Path(__file__).resolve().parent
+DATA_DIR = BASE_DIR / 'data'  # directory to store our recipe text files
+EMBEDDING_MODEL = 'text-similarity-ada-001'  # OpenAI embedding model
+PINECONE_INDEX_NAME = 'recipes'  # name of our Pinecone index
+```
+</CodeGroup>
+
+Next, create a `openai_utils.py` file. This includes a `get_vector_for_text` function that allows us to pass in text
+and responds with a vector. We will later store these vectors in Pinecone.
+
+<CodeGroup>
+```python openai_utils.py
+import openai
+
+from settings import EMBEDDING_MODEL
+
+def get_vector_for_text(text):
+    response = openai.Embedding.create(
+        input=text,
+        engine=EMBEDDING_MODEL,
+    )
+    return response['data'][0]['embedding']
+```
+</CodeGroup>
+
+Now create a new file called `io_utils.py`. This file includes a helper function called `get_recipe_filepaths` to
+retrieve a list of recipe file paths.
+
+<CodeGroup>
+```python io_utils.py
+import os
+
+from settings import DATA_DIR
+
+def get_recipe_filepaths():
+    results = []
+
+    for file in os.listdir(DATA_DIR):
+        if file.endswith('.txt'):
+            results.append(DATA_DIR / file)
+
+    return results
+```
+</CodeGroup>
+
+Next, create a `files.py` file that contains the functionality used to parse file names and read text files. The
```

Comment:
Why are `files.py` and `io_utils.py` separated here? IMO adds complexity to tutorial without any value to user